src/openassetio-python/package/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -968,6 +968,262 @@ def __pager_page_iter(pager):
             pager.next()
 
 
+class Test_getWithRelationshipsPaged(FixtureAugmentedTestCase):
```

Comment:
Yeah the lack of DRYness does bother me.  I worry less with tests, since immediate readability trumps DRYness for tests.  But the repetition is a bit extreme.

We could break the rule of one class per method, and do all methods in a single `TestRelationships` test class, e.g. (some refactored version of):

```python
    def test_when_relation_unknown_then_empty_list_returned(self):
        a_ref = self.requireEntityReferenceFixture("a_reference", skipTestIfMissing=True)
        an_unknown_rel = TraitsData({"🐠🐟🐠🐟"})

        self.__testGetWithRelationship([a_ref], an_unknown_rel, [[]])

        self.__testGetWithRelationships(a_ref, [an_unknown_rel], [[]])

        [pager] = self.__test_getWithRelationshipPaged_success([a_ref], an_unknown_rel)
        self._assert_pager_is_at_end(pager)

        [pager] = self.__test_getWithRelationshipsPaged_success(a_ref, [an_unknown_rel])
        self._assert_pager_is_at_end(pager)
```

Reply:
I think thats a total winner, as they key key thing we really need to assert is that the business logic is consistent between the methods, and separate fixtures greatly increases the chance that people don't realize this is the case?

Reply:
Is this something we should do as part of this PR?

Reply:
I guess it would change the fixtures for someone who uses these checks right? If not in this PR, sooner would be better than later?

Reply:
This does also fix the " if you implement one you have to implement them all" thing too doesn't it.

Reply:
I'll do it as part of this PR (converted to Draft in the meantime)

Reply:
Merged the tests in e66dc64c

Reply:
What a fabulous refactor. Resolving on behalf of Tom who's in training.